core/types/gen_receipt_json.go

```diff
@@ -25,7 +25,7 @@ func (r Receipt) MarshalJSON() ([]byte, error) {
 		TxHash            common.Hash    `json:"transactionHash" gencodec:"required"`
 		ContractAddress   common.Address `json:"contractAddress"`
 		GasUsed           hexutil.Uint64 `json:"gasUsed" gencodec:"required"`
-		EffectiveGasPrice *hexutil.Big   `json:"effectiveGasPrice,omitempty"`
+		EffectiveGasPrice *big.Int       `json:"effectiveGasPrice"`
```

Comment:
https://github.com/ethereum/go-ethereum/commit/87186148e0c4e46f7825e01cdcc57e833d1d3be6
> This change adds a struct field EffectiveGasPrice in types.Receipt. The field is present
in RPC responses, but not in the Go struct, and thus can't easily be accessed via ethclient

This change is related to an error I'm seeing locally at `TestT8n`, where the `effectiveGasPrice: null` pair is present but unexpected. 
This branch has recently done a `go generate ./...` command.

Simply adding `,omitempty` to the `gen_receipt` `MarshalJSON` method will fix it, but seems again like there might be a better way than editing the generated file... but in other Receipt contexts it seems we're identical to upstream. 

- https://github.com/ethereum/go-ethereum/commit/87186148e0c4e46f7825e01cdcc57e833d1d3be6#diff-2e925d78462a4abc9f45e63fdaa56401becf8dce096ac72297e1adc85511e02cR28